apps/designer/app/shared/instance-utils.ts

```diff
@@ -0,0 +1,44 @@
+import store from "immerhin";
+import { Instance } from "@webstudio-is/project-build";
+import {
+  rootInstanceContainer,
+  propsStore,
+  stylesStore,
+  selectedInstanceIdStore,
+} from "./nano-states";
+import { findSubtree } from "./tree-utils";
+import { removeByMutable } from "./array-utils";
+
+export const deleteInstance = (targetInstanceId: Instance["id"]) => {
```

Comment:
- we have been intentionally decoupling a mutable utility function  from store stuff and calling it with Mutate suffix, so we can test them in isolation

- its weird that you now have both, deleteInstance and deleteInstanceMutable 
- need unit tests for every non-trivial utility we write, this one seems untested

Reply:
It is trivial. Non trivial is traversing of instances. Here we just ditch all matching data with tested utility.

Yeah, deleteInstanceMutable no longer necessary.

Reply:
I guess trivial is relative, in this case its doing many operations and more will be added, so I would test it, because its a very important one

Reply:
I would extract the mutable part of this function and write tests using it

Reply:
could be a snapshot test if its too much data

Reply:
as for immutable deleteInstance that wrapps the transition, are we using it in many places? if not I would just inline the transaction where its used

Reply:
We gonna use it instead of pubsub.

Reply:
> could be a snapshot test if its too much data

even writing input is a tough call